### ISSUE TITLE
Google escapes the container

### ISSUE THREAD

**Issue by 7ider5** [mozilla/multi-account-containers#2409]
### Before submitting a bug report

- [X] I updated to the latest version of Multi-Account Container and tested if I can reproduce the issue
- [X] I searched for existing reports to see if it hasn't already been reported

### Step to reproduce

1. Create a "google" container
2. Open the "google" container
3. Type >mail.google.com and press enter

### Actual behavior

The page redirects to https://www.google.com/intl/fr/gmail/about/ and the container switches to a classic tab.
![Escape](https://user-images.githubusercontent.com/10852205/187643113-0c85299b-61d6-485c-a96a-8b2de8495f54.gif)

### Expected behavior

The redirection must stay in the container

### Additional informations

I tried a workaround creating "always open in container" rules with the domains google.com, maill.google.com and accounts.google.com. In that case the tab seems to stay in the container, but google login returns an error 400.

Multi-account containers version : 8.0.7
Firefox version : 104.0
Windows 21H2

**Comment by ahahahahahahahahahahahahaha** [comment]
Looks like 8.0.5-6 the most unsuccessful versions, because in addition they have a critical bug that completly ignores http proxy (but this works in previous versions).

I tried to reproduce your bug in 8.0.4 version and everything worked perfectly. Moreover I added rule "always open in container" for mail.google.com and I successfully logged in and the site opens without problems in "google" container.

Win 7, Firefox 104.0.2
